libs/geometry/src/TangentSpaceMesh.cpp

```diff
@@ -87,21 +86,21 @@ std::string_view to_string(Algorithm algorithm) {
             return "FRISVAD";
         case Algorithm::FLAT_SHADING:
             return "FLAT_SHADING";
-        case Algorithm::SIGN_OF_W:
-            return "SIGN_OF_W";
         default:
             PANIC_POSTCONDITION("Unknown algorithm %u", static_cast<uint8_t>(algorithm));
     }
 }
 
-Algorithm selectBestDefaultAlgorithm(uint8_t inputType) {
+inline bool isInputType(const uint8_t inputType, const uint8_t checkType) {
+    return ((inputType & checkType) == checkType);
+}
+
+Algorithm selectBestDefaultAlgorithm(uint8_t inputType) noexcept {
```

Comment:
can't be `noexcept` because you have `ASSERT_PRECONDITION`.
From a style standpoint we always try to have the precondition asserts be the first thing the method does -- or have them just before calling the method.

Reply:
done